docs/spec/distributing.rst

```diff
@@ -10,72 +10,428 @@ Stub files
+not yet supported by all major type checkers. Unless otherwise noted in this
+tracker, stub files support all features from the ``typing`` module of the
+latest released Python version. If a stub uses typing features from a later
+Python version than what the implementation supports, these features can be
+imported from ``typing_extensions`` instead of ``typing``
+
+For example, a stub could use ``Literal``, introduced in Python 3.8,
+for a library supporting Python 3.7+::
+
+    from typing_extensions import Literal
+
+    def foo(x: Literal[""]) -> int: ...
+
+Comments
+""""""""
+
+Standard Python comments are accepted everywhere Python syntax allows them.
+
+Two kinds of structured comments are accepted:
+
+* A ``# type: X`` comment at the end of a line that defines a variable,
+  declaring that the variable has type ``X``. However, :pep:`526`-style
+  variable annotations are preferred over type comments.
+* An error suppression comment at the end of any line. Common error suppression
+  formats are ``# type: ignore``, ``# type: ignore[error-class]``, and
+  ``# pyright: ignore[error-class]``. Type checkers may ignore error
+  suppressions that they don't support but should not error on them.
+
+Imports
+"""""""
+
+Stub files distinguish between imports that are re-exported and those
+that are only used internally. See :ref:`import-conventions`.
+
+Type aliases can be used to re-export an import under a different name::
+
+    from foo import bar as _bar
+    new_bar = _bar  # "bar" gets re-exported with the name "new_bar"
+
+Stubs support customizing star import semantics by defining a module-level
+variable called ``__all__``. In stubs, this must be a string list literal.
+Other types are not supported. Neither is the dynamic creation of this
+variable (for example by concatenation).
+
+When ``__all__`` is defined, exactly those names specified in ``__all__`` are
+imported::
+
+    __all__ = ['public_attr', '_private_looking_public_attr']
+
+    public_attr: int
+    _private_looking_public_attr: int
+    private_attr: int
+
+Type checkers support cyclic imports in stub files.
+
+Built-in Generics
+"""""""""""""""""
+
+:pep:`585` built-in generics are supported and should be used instead
+of the corresponding types from ``typing``::
+
+    from collections import defaultdict
+
+    def foo(t: type[MyClass]) -> list[int]: ...
+    x: defaultdict[int]
+
+Using imports from ``collections.abc`` instead of ``typing`` is
+generally possible and recommended::
 
-* Extension modules
+    from collections.abc import Iterable
 
-* Third-party modules whose authors have not yet added type hints
+    def foo(iter: Iterable[int]) -> None: ...
 
-* Standard library modules for which type hints have not yet been
-  written
+Unions
+""""""
+
+Declaring unions with the shorthand syntax or ``Union`` and ``Optional`` is
```

Comment:
Likewise, I don't think this belongs in the typing spec. It's fine content for a guide.